src/Common/TraceSender.cpp

```diff
@@ -42,13 +42,13 @@ void TraceSender::send(TraceType trace_type, const StackTrace & stack_trace, Int
     char buffer[buf_size];
     WriteBufferFromFileDescriptorDiscardOnFailure out(pipe.fds_rw[1], buf_size, buffer);
 
-    StringRef query_id;
+    std::string_view query_id;
     UInt64 thread_id;
 
     if (CurrentThread::isInitialized())
     {
-        query_id = StringRef(CurrentThread::getQueryId());
-        query_id.size = std::min(query_id.size, QUERY_ID_MAX_LEN);
+        query_id = CurrentThread::getQueryId();
+        query_id = std::string_view(query_id.data(), std::min(query_id.size(), QUERY_ID_MAX_LEN));
```

Comment:
These two lines look strange.

Reply:
Previously we could just change the size of the StringRef to shrink it, but we cannot assign the size to string_view directly. As an option this could be done:
```suggestion
        if (query_id.size() > QUERY_ID_MAX_LEN)
            query_id.remove_suffix(query_id.size() - QUERY_ID_MAX_LEN);
```